pkg/controller/ecr/repository/controller.go

```diff
@@ -0,0 +1,281 @@
+	"context"
+	"sort"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsecr "github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/ecr/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+	ecr "github.com/crossplane/provider-aws/pkg/clients/ecr"
+)
+
+const (
+	errUnexpectedObject = "The managed resource is not an repository resource"
+	errKubeUpdateFailed = "cannot update repository custom resource"
+
+	errDescribe            = "failed to describe repository with id"
+	errMultipleItems       = "retrieved multiple repository for the given ECR name"
+	errCreate              = "failed to create the repository resource"
+	errCreateTags          = "failed to create tags for the repository resource"
+	errRemoveTags          = "failed to remove tags for the repository resource"
+	errListTags            = "failed to list tags for the repository resource"
+	errDelete              = "failed to delete the repository resource"
+	errSpecUpdate          = "cannot update spec of repository custom resource"
+	errStatusUpdate        = "cannot update status of repository custom resource"
+	errUpToDateFailed      = "cannot check whether object is up-to-date"
+	errUpdateScan          = "failed to update scan config for repository resource"
+	errUpdateMutability    = "failed to update mutability for repository resource"
+	errPatchCreationFailed = "cannot create a patch object"
+)
+
+// SetupRepository adds a controller that reconciles ECR.
+func SetupRepository(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.RepositoryGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Repository{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.RepositoryGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(managed.NewNameAsExternalName(mgr.GetClient()), &tagger{kube: mgr.GetClient()}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cfg, err := awsclients.GetConfig(ctx, c.kube, mg, "")
```

Comment:
Is `Repository` global or scoped to a region? If the former we should pass `"aws-global"` here, if the latter we should add `region` field to spec and pass it here 👍 (this is brand new, see https://github.com/crossplane/provider-aws/pull/339 for more info)